include/OnnxMlirRuntime.h

```diff
@@ -108,9 +108,19 @@
  * \subsection reference Reference
  *
  * For full reference to available C Runtime API, refer to
- *`include/onnx-mlir/Runtime/OMTensor.h` and
+ * `include/onnx-mlir/Runtime/OMTensor.h` and
  * `include/onnx-mlir/Runtime/OMTensorList.h`.
  *
  */
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern OMTensorList* run_main_graph(OMTensorList*);
```

Comment:
Hmm...I thought `run_main_graph` is part of the user API, no? Clearly the user needs to be told how to call the model entry point. Maybe I'm missing something?

Reply:
It's documented in the intro/tutorial, but not as part of the API and it appears to me that this information should not be in the API spec for onnx-mlir, as it stipulates the API of the compiler output, not the compiler per se. Obviously, the user must know this piece of information to call the model, and my thought is to let the user declare it themselves. Benefits as listed above. Also, I think users should have the freedom to rename this entry point symbol, in the event that they want to embed multiple models in the same application.

Reply:
OK so I guess I obviously misunderstood something. So you are saying the users can actually decide for themselves what the model entry point will be called? If that's the case then clearly the prototype doesn't belong in OnnxMlirRuntime.h and I will take it out. But I'm curious, how would a user generate an entry point of his/her own instead of run_main_graph? I don't see an external mechanism for a user to do that without changing onnx-mlir itself, or this will be added later?

Reply:
> OK so I guess I obviously misunderstood something. So you are saying the users can actually decide for themselves what the model entry point will be called?

I don't think you're mistaken, it's not a feature we support today (although sophisticated users can do so even today via `objcopy` to rename symbols). I'm simply brainstorming the implications of having this extern definition in the API header, and it occurs to me that one of them is that it'll complicate things when using multiple models in a single application. 

> But I'm curious, how would a user generate an entry point of his/her own instead of run_main_graph? I don't see an external mechanism for a user to do that without changing onnx-mlir itself, or this will be added later?

Not sure if we want to support it officially, but as I mentioned before, renaming symbols can be somewhat easily done with `objcopy`. We actually do it today with embedded constant pool here: https://github.com/onnx/onnx-mlir/blob/414d702ac4909c657eadb19340d08506272785b5/src/MainUtils.cpp#L243 .

Reply:
Yeah I remember the objcopy although I think it's probably a bit hacky and may not be very portable. In any case, I think this is also getting into the developer territory and I don't expect unsophisticated users wanting to touch it.

The best solution still seems to be having a reasonable default like run_main_graph but allowing sophisticated users to override that. Perhaps one option is to generate a companion model.h which defaults to the run_main_graph prototype declaration but can be customized somehow by the user (like through some emit option?). This is basically what JNI does. When you declare a native method in Java, you can invoke javah on the compiled class to get the prototype for the native function you must implement.